DataManagementSystem/Client/FTS3Job.py

```diff
@@ -266,6 +300,39 @@ def _constructTransferJob(self, pinTime, allLFNs, target_spacetoken, protocols=N
 
     allSrcDstSURLs = res['Value']['Successful']
 
+    # This contains the staging URLs if they are different from the transfer URLs
+    # (CTA...)
+    allStageURLs = dict()
+
+    # In case we are transfering from a tape system, and the stage protocol
+    # is not the same as the transfer protocol, we generate the staging URLs
+    # to do a multihop transfer. See bellow.
+    if sourceIsTape:
+      srcProto, _destProto = res['Value']['Protocols']
+      if srcProto not in srcSE.localStageProtocolList:
```

Comment:
I asked @chaen directly what srcProto and localStageProtocolList is for CTA, because I got lost